code/model/UserDefinedForm.php

```diff
@@ -1377,9 +1377,25 @@ public function edit($request) {
 	 */
 	public function preview() {
 		return $this->customise(new ArrayData(array(
-			'Body' => $this->record->getEmailBodyContent()
+			'Body' => $this->record->getEmailBodyContent(),
+			'HideFormData' => $this->record->HideFormData,
+			'Fields' => $this->getPreviewFieldData()
 		)))->renderWith($this->record->EmailTemplate);
 	}
+
+	/**
+	 * Get some placeholder field values to display in the preview
+	 * @return ArrayList
+	 */
+	private function getPreviewFieldData() {
+		$data = new ArrayList();
+
+		foreach ($this->record->Form()->Fields() as $field) {
+			$data->push(DBField::create_field('Text', $field->Name, '$' . $field->Name));
```

Comment:
You'll need to ensure that the format of the item in this list matches the same list of fields as in the normal field. The necessary fields are 'Name', 'Title', 'Value' and 'FormattedValue'.

```php
$data->push(new ArrayData(array(
    'Name' => $field->Name,
    'Title' => $field->Title,
    'Value' => '$' . $field->Name,
    'SubmittedValue' => '$' . $field->Name
)));
```

I would also suggest to iterate through `$record->Fields()` (which is the list of `EditableFormField` instances) rather than generating the form and selecting the `FormField` instances from there.